sdk/core/core-http/package.json

```diff
@@ -17,16 +13,21 @@
     "autorest",
     "clientruntime"
   ],
+  "engines": {
+    "node": ">=8.0.0"
+  },
   "keywords": [
     "isomorphic",
     "browser",
     "javascript",
     "node",
     "microsoft",
     "autorest",
-    "clientruntime"
+    "clientruntime",
+    "Azure",
+    "cloud"
   ],
-  "main": "./dist/coreHttp.node.js",
+  "main": "dist/index.js",
```

Comment:
This change is ok, I packed it and checked. The file is there, and the contents are reasonable.

Reply:
What is the motivation for changing `main` to use `index.js`? Wouldnt the same reason apply to the `module` field as well?

Reply:
And how does the changes in #9380 affect this? cc @chradek

Reply:
> What is the motivation for changing `main` to use `index.js`? Wouldnt the same reason apply to the `module` field as well?

Here's the eslint-rule error that gets logged when I revert this:

```
  34:11  error  main is set to ./dist/coreHttp.node.js when it should be set to dist/index.js  @azure/azure-sdk/ts-package-json-main-is-cjs
```

I think I need to summon @willmtemple in order to have a better response here.

Reply:
The linting rules assume that `dist/index.js` should be the entry point for the package and that rollup should be configured to output its bundle there. For some core packages, that might not be true, so it can be disabled if necessary. (I remember that we disabled dtsRollup in core-http a while ago due to a bug in API Extractor.)

The rollup output path is also changed, so I think this should be fine. The ESM should still be in the same place, so the `module` field won't need to be updated. We should try to run this new version through webpack to make sure it bundles correctly as well.